### DISCUSSION TITLE
System date time vs User date time

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8127] [Help] [answered]
Hi all,

If I understand well {{ date now }} binding is returning the date and time of the FE user terminal connected to Budibase server.
Is there a way to get the date and time of the server running Budibase?
What we are interested in is having a way to have a common clock for users connecting from different machines and regions. 
So I would like to store the system UTC time for user events and not the browser time.
We are using MYSQL as database running on same host as Budibase and probably we could use a` select utc_timestamp` query or triggers to add utc_timestamp at inserts and updates , but I would like to know if there any other options available.

Please help!

**Comment by andz-bb** [comment, marked as answer]
Hey @gmentzik 

If you trust the date on the client machine but just want a consistent timezone then you could use JavaScript bindings instead of Handlebars to return a UTC timestamp. Something like `return new Date().toUTCString();` would work here and will always give you a UTC time based on the client's machine regardless of their timezone.

If you would prefer to get the time from a server I would suggest adding a query to your MySQL datasource and getting the time from that. `SELECT NOW()` worked perfectly on mine. I'm not sure how to get your Budibase server's time within the builder but I think the DB server should work fine for you. If you have any questions on this I'm happy to help 👍

**Reply by gmentzik** [reply]
Hi @andz-bb ,
Thank you for your answer. 
I am using binding {{ date now "YYYY-MM-DD hh:mm:ss" UTC }} to get FE time in UTC and format it as Mysql datetime field requires it. My guess is that behind the scenes it does same thing as the Javascript you suggest and anyhow both these methods give back FE time. My question is if Budibase has any way to provide server time. When developing application where a common clock reference is needed it is usual to get server time from the server backend services are running or from some clock service. If not possible, as you say my only way around is to query it from MySQL server.
